From 6226b1dd91287e986dde2d4f81cf6f9205dc21c0 Mon Sep 17 00:00:00 2001
From: nikhilad <nikhilad@ami.com>
Date: Wed, 29 Oct 2025 17:45:24 +0530
Subject: [PATCH] Added Fix For SDR Preserve Configuration.

[Recreated the patch]

Description:
SDR will not be preserved when PreserveConfiguration SDR is true

Tested:
Build taken successfully and Verified.

Working as Expected.

Signed-off-by: saravananm <saravananm@ami.com>

Signed-off-by: nikhilad <nikhilad@ami.com>
---
 src/entity_manager/entity_manager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/entity_manager/entity_manager.cpp b/src/entity_manager/entity_manager.cpp
index 6c48769..ea3acd6 100644
--- a/src/entity_manager/entity_manager.cpp
+++ b/src/entity_manager/entity_manager.cpp
@@ -681,7 +681,7 @@ int main()
     });
     dbus_interface::tryIfaceInitialize(entityIface);
 
-    if (em_utils::fwVersionIsSame())
+    if (em_utils::fwVersionIsSame() || std::filesystem::exists(configuration::currentConfiguration))
     {
         if (std::filesystem::is_regular_file(
                 configuration::currentConfiguration))
-- 
2.34.1

